src/components/lib/component/arch/x86_64/cos_asm_simple_stacks.h

```diff
@@ -25,6 +25,57 @@
 
 #define COS_ASM_REQUEST_STACK
 
+#define COS_ULINV_SWITCH_DOMAIN(protdom)			\
+	movl	$protdom,  %ecx;				\
+	addl	%ecx,      %ecx;				\
+	movl	$0b11,     %eax;				\
+	sall	%cl,       %eax;				\
+	notl	%eax;						\
+	andl	$-4,       %eax;				\
```

Comment:
@gparmer https://www.felixcloutier.com/x86/wrpkru Some useful infomation to know. Using 32 bit registers seems to be just because of the requirement of wrpkru